src/traces/scatterregl/plot.js

```diff
@@ -0,0 +1,575 @@
+
+            isOpen = /-open/.test(symbol);
+            isDot = /-dot/.test(symbol);
+            isDimmed = selIds && !selIds[index];
+
+            _colors = colors;
+
+            if(isOpen) {
+                _borderColors = colors;
+            } else {
+                _borderColors = borderColors;
+            }
+
+            // See  https://github.com/plotly/plotly.js/pull/1781#discussion_r121820798
+            // for more info on this logic
+            size = sizes[index];
+            bw = borderSizes[index];
+
+            this.scatterOptions.sizes[i] = size;
+
+            if(symbol === 'circle') {
+                this.scatterOptions.markers[i] = null;
+            }
+            else {
+                // get symbol sdf from cache or generate it
+                if(SYMBOL_SDF[symbol]) {
+                    symbolSdf = SYMBOL_SDF[symbol];
+                } else {
+                    if(isDot && !symbolNoDot) {
+                        symbolPath = symbolFunc(SYMBOL_SIZE * 1.1) + SYMBOL_SVG_CIRCLE;
+                    }
+                    else {
+                        symbolPath = symbolFunc(SYMBOL_SIZE);
+                    }
+
+                    symbolSdf = svgSdf(symbolPath, {
+                        w: SYMBOL_SDF_SIZE,
+                        h: SYMBOL_SDF_SIZE,
+                        viewBox: [-SYMBOL_SIZE, -SYMBOL_SIZE, SYMBOL_SIZE, SYMBOL_SIZE],
+                        stroke: symbolNoFill ? SYMBOL_STROKE : -SYMBOL_STROKE
+                    });
+                    SYMBOL_SDF[symbol] = symbolSdf;
+                }
+
+                this.scatterOptions.markers[i] = symbolSdf || null;
+            }
+            this.scatterOptions.borderSizes[i] = 0.5 * bw;
+
+            var optColors = this.scatterOptions.colors;
+            var dim = isDimmed ? DESELECTDIM : 1;
+            if(!optColors[i]) optColors[i] = [];
+            if(isOpen || symbolNoFill) {
+                optColors[i][0] = TRANSPARENT[0];
+                optColors[i][1] = TRANSPARENT[1];
+                optColors[i][2] = TRANSPARENT[2];
+                optColors[i][3] = TRANSPARENT[3];
+            } else {
+                optColors[i][0] = _colors[4 * index + 0] * 255;
+                optColors[i][1] = _colors[4 * index + 1] * 255;
+                optColors[i][2] = _colors[4 * index + 2] * 255;
+                optColors[i][3] = dim * _colors[4 * index + 3] * 255;
+            }
+            if(!this.scatterOptions.borderColors[i]) this.scatterOptions.borderColors[i] = [];
+            this.scatterOptions.borderColors[i][0] = _borderColors[4 * index + 0] * 255;
+            this.scatterOptions.borderColors[i][1] = _borderColors[4 * index + 1] * 255;
+            this.scatterOptions.borderColors[i][2] = _borderColors[4 * index + 2] * 255;
+            this.scatterOptions.borderColors[i][3] = dim * _borderColors[4 * index + 3] * 255;
+        }
+
+        // prevent scatter from resnapping points
+        this.regl._refresh();
```

Comment:
So you have `regl` "scene-wide" commands inside a trace module?

Can you factor those out so that all gl traces can _just work_?

Reply:
I understand the most of the `plots/gl2d/scene2d/` logic will be :hocho: as we'll now reuse many cartesian axes features, but we'll still need some subplot-wide logic, and that should still be placed in `plots/gl2d/scene2d.js` so that all current (and future) gl2d trace module can benefit from it.

Reply:
Hmm… is this a manual override? `_refresh` is the unsafe escape hatch, right? https://github.com/regl-project/regl/blob/gh-pages/API.md#unsafe-escape-hatch

Reply:
Ok, makes sense, I'll just put `regl._refresh` into regl-* components.

Reply:
@rreusser see https://github.com/regl-project/regl/issues/427

Reply:
Got it! Thought maybe that was the source.